contrib/performance/performance_pkg/graph/config.py.in

```diff
@@ -0,0 +1,233 @@
+# LANL.
+#
+# THIS SOFTWARE IS PROVIDED BY LOS ALAMOS NATIONAL SECURITY, LLC AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL LOS ALAMOS NATIONAL SECURITY, LLC OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+# OF SUCH DAMAGE.
+
+
+
+from configparser import ConfigParser, ExtendedInterpolation
+from matplotlib import pyplot as plt # should be able to get rid of this
+
+from performance_pkg import common
+
+# config sections and keys
+RAW_DATA = 'raw_data'                   # section
+RAW_DATA_COMMITS = 'commits'            # string list
+RAW_DATA_COLUMNS = 'columns'            # string list
+
+OUTPUT = 'output'                       # section
+OUTPUT_PATH = 'path'                    # string
+OUTPUT_GRAPH_TITLE = 'graph_title'      # string
+OUTPUT_DIMENSIONS = 'graph_dimensions'  # integer pair
+
+LINES = 'lines'                         # section
+LINES_COLORS = 'colors'                 # string list
+LINES_TYPES = 'types'                   # string list
+LINES_MARKERS = 'markers'               # string list
+
+AXES = 'axes'                           # section
+AXES_HASH_LEN = 'hash_len'              # non-zero integer
+AXES_Y_LABEL = 'y_label'                # string
+AXES_Y_MIN = 'y_min'                    # number
+AXES_Y_MAX = 'y_max'                    # number
+
+ANNOTATIONS = 'annotations'             # section
+ANNOTATIONS_SHOW = 'show'               # bool
+ANNOTATIONS_PRECISION = 'precision'     # positive integer
+ANNOTATIONS_X_OFFSET = 'x_offset'       # number
+ANNOTATIONS_Y_OFFSET = 'y_offset'       # number
+ANNOTATIONS_TEXT_COLORS = 'text_colors' # string list
+
+ERROR_BAR = 'error_bar'                 # section
+ERROR_BAR_SHOW = 'show'                 # bool
+ERROR_BAR_PRECISION = 'precision'       # positive integer
+ERROR_BAR_MIN_MAX = 'min_max'           # bool
+ERROR_BAR_TEXT_COLORS = 'text_colors'   # string list
+
+def to_list(value, convert=str):
+    return [convert(item.strip()) for item in value.split(',')]
+
+def str_list(value):
+    return to_list(value)
+
+def int_list(value):
+    return to_list(value, int)
+
+def pos_int_list(value):
+    values = to_list(value, int)
+    for val in values:
+        if val < 1:
+            raise ValueError('Positive integer list got a non-positive number: {0}'.format(val))
+    return values
+
+def read_value(parser, section, key,
```

Comment:
entry? this function only reads config values. theres no need to add `config` to the name because it is only ever called in the config script. this function should really be private to this file